packages/block-library/src/gallery/edit.js

```diff
@@ -355,11 +355,10 @@ class GalleryEdit extends Component {
 		} = attributes;
 
 		const hasImages = !! images.length;
-		const hasImagesWithId = hasImages && some( images, ( { id } ) => id );
 
 		const mediaPlaceholder = (
 			<MediaPlaceholder
-				addToGallery={ hasImagesWithId }
+				addToGallery={ true }
```

Comment:
It looks like this might is the cause of https://github.com/WordPress/gutenberg/issues/22338, that when first adding a gallery block the Media Library is always in `addToGallery` mode with all images selected on the Insert Gallery page.

Reply:
I've read more of the details on this PR, and I see that the issue is related to `hasImagesWithId` being false when an image upload is in progress and other operations causing the uploading image to be replaced.

I've made naive PR at https://github.com/WordPress/gutenberg/pull/22659.